assets/js/state/softwareUpdatesSettings.js

```diff
@@ -0,0 +1,50 @@
+import { createAction, createSlice } from '@reduxjs/toolkit';
+
+export const UPDATE_CATALOG = 'UPDATE_CATALOG';
+export const updateCatalog = createAction(UPDATE_CATALOG);
+
+const emptySettings = {
+  url: undefined,
+  username: undefined,
+  ca_uploaded_at: undefined,
+};
+
+const initialState = {
+  loading: false,
+  settings: emptySettings,
+  error: null,
+};
+
+export const softwareUpdatesSettingsSlice = createSlice({
+  name: 'softwareUpdatesSettings',
+  initialState,
+  reducers: {
+    startLoadingSoftwareUpdatesSettings: (state) => {
+      state.loading = true;
+    },
+    setSoftwareUpdatesSettings: (state, { payload: { settings } }) => {
+      state.loading = false;
+      state.error = null;
+      state.settings = settings;
+    },
+    setEmptySoftwareUpdatesSettings: (state) => {
+      state.loading = false;
+      state.error = null;
+      state.settings = emptySettings;
+    },
+  },
+});
```

Comment:
I would prefer a more granular structure that allows us to stop loading independently but it's absolutely out of personal preference

Reply:
Only thing that comes to my mind is that in a saga we might end up in unnecessary repetition imho, for instance we might be doing something like this when handing errors.
```js
try {
  // call the endpoint
  setSoftwareUpdatesSettings(settings);
  stopLoadingSoftwareUpdatesSettings();
} catch (error) {
  setEmptySoftwareUpdatesSettings()
  stopLoadingSoftwareUpdatesSettings();
}
```

Let's keep an eye on that anyway and see if an independent `stopLoadingSoftwareUpdatesSettings` might be helpful.

Reply:
No problem about that, absolutely :+1:

Reply:
partially true what I said, we could actually have something like this
```js
try {
  // call the endpoint
  setSoftwareUpdatesSettings(settings);
} catch (error) {
  setEmptySoftwareUpdatesSettings();
}
stopLoadingSoftwareUpdatesSettings();
```
Anyway, keeping the state mutation internally in the slice, feels more robust. If we need to change, we'll change.